### PR TITLE
GSW-1839 refactor: use avl.Tree in position

### PR DESCRIPTION
**Pull request by r3v4s**
## refactor
1. use `avl.Tree` instead of map
- kept `unit64` type for tokenId (no need to use seqid type for it)

image

### CODE REVIEW

**Review by notJoon — CHANGES_REQUESTED** [review]
It looks good. I added some comments for the documentation things, please check them.

File: position/position.gno [hunk 1]
```diff
@@ -776,20 +799,20 @@ func updateTokensOwed(
...
+	position := MustGetPosition(tokenId)
 
 	if !(position.isClear()) {
 		panic(addDetailToError(
 			errNotClear,
 			ufmt.Sprintf("position.gno__burnNFT() || position(%d) isn't clear(liquidity:%s, tokensOwed0:%s, tokensOwed1:%s)", tokenId, position.liquidity.ToString(), position.tokensOwed0.ToString(), position.tokensOwed1.ToString()),
 		))
 	}
-	delete(positions, tokenId)
+	removePosition(tokenId)
```
[the remark this answers is not in the document]
Comment (dongwon8247) [reply position/position.gno:810]: fyi - we're removing the `file_func name` in the error/panic message like `position.gno__burnNFT()` here.

File: position/position.gno [hunk 2]
```diff
@@ -343,7 +352,11 @@ func increaseLiquidity(params IncreaseLiquidityParams) (uint64, *u256.Uint, *u25
 	position.liquidity = new(u256.Uint).Add(position.liquidity, liquidity)
 	position.burned = false
 
-	positions[params.tokenId] = position
+	updated := setPosition(params.tokenId, position)
+	if !updated {
+		// TODO:
```
[the remark this answers is not in the document]
Comment (r3v4s) [reply position/position.gno]: oh sorry, missed it.
It was meant to add new error type for hard coded error string
34e10148d280714a5902fe384923c02f0ecba81d

File: position/position.gno (hunk 1 above)
[the remark this answers is not in the document]
Comment (r3v4s) [reply position/position.gno]: removed unused function
removed `filename__function()` template in all error message string

529fd2fedde9d10f1b73359533825f152c339900

**Review by notJoon — DISMISSED** [review]
LGTM 👍

**Review by onlyhyde — APPROVED** [review]

**Review by notJoon — APPROVED** [review]

**Pull request merged**